src/tagmanager/tm_workspace.c

```diff
@@ -828,20 +828,22 @@ static GHashTable *get_includes(TMSourceFile *source, GPtrArray **header_candida
 		gchar *hdr_name = g_path_get_basename(hdr_tag->name);
 		GPtrArray *tm_files = g_hash_table_lookup(theWorkspace->source_file_map, hdr_name);
 
-		if (tm_files)
+		if (tm_files && tm_files->len > 0)
 		{
 			guint j;
 
-			for (j = 0; j < tm_files->len; j++)
+			if (!*header_candidates)
 			{
-				TMSourceFile *hdr = tm_files->pdata[j];
+				TMSourceFile *hdr = tm_files->pdata[0];
 				gchar **hdr_comps = g_strsplit(hdr->short_name, ".", 2);
```

Comment:
I see this only now. Doesn't that break on files with multiple dots in name? I thought we want to compare the base name without extension, i.e. up until the last dot.

Reply:
Yes, you are right. Well, I could `g_strdup()` the string, use strrchr() to find the `.` and replace it with `\0` which would give us the base name. Or is there some simpler solution?

Reply:
I think I would use `g_path_get_basename()` instead of `g_strdup()`. But it doesn't matter much so yea I don't know a simpler solution.

Reply:
IIUC the intention here is to get the filename without the last extension, `g_path_get_basename()` does not do that.  But `utils_remove_ext_from_filename()` does.

Note that there are other `strsplit()`s above that may need changing.

Reply:
I know that. `g_path_get_basename()` (or `g_strdup()`) must be combined with `strrchr()` and setting the last dot to `NUL` as per @techee suggestion. `utils_remove_ext_from_filename()` does that and is a good candidate (though I'm not sure if @techee wants to use Geany utility function in tagmanager code).

Reply:
> I'm not sure if @techee wants to use Geany utility function in tagmanager code

AFAIK there is no "upstream" for tagmanager, and this PR is modifying it anyway, so why not.